iterations_runners/iterations_runner.lua

```diff
@@ -31,15 +28,14 @@ for BM_i = 1, BM_iters, 1 do -- inclusive upper bound in lua
     run_iter(BM_param) -- run one iteration of benchmark
     local BM_end_time = kruntime.clock_gettime_monotonic()
 
-    local BM_intvl = BM_end_time - BM_start_time
+    BM_iter_times[BM_i] = BM_end_time - BM_start_time
+end
 
-    io.stdout:write(BM_intvl)
+io.stdout:write("[")
+for BM_i = 1, BM_iters, 1 do
+    io.stdout:write(BM_iter_times[BM_i])
     if BM_i < BM_iters then
         io.stdout:write(", ")
     end
-
-    io.stdout:flush()
 end
-
-io.stdout:write("]")
-io.stdout:flush()
+print("]")
```

Comment:
print has a newline suffixed, io.stdout:write does not. Where possible I tried to keep all output on one line. It doesn't really matter since all benchmarking is done by the time we print.